numpy/core/shape_base.py

```diff
@@ -348,3 +353,83 @@ def stack(arrays, axis=0):
     sl = (slice(None),) * axis + (_nx.newaxis,)
     expanded_arrays = [arr[sl] for arr in arrays]
     return _nx.concatenate(expanded_arrays, axis=axis)
+
+
+def block(tup_tup):
+    """
+    Create block arrays similar to Matlab's "square bracket stacking":
+
+            [A A; B B]
+
+    You can create a block array with the same notation you use for
+    `np.array`.
+
+    Parameters
+    ----------
+    tup_tup : sequence of sequence of ndarrays
```

Comment:
sequence of sequence is quite ugly. Looking at the examples, why would it be needed to use a list of a single array, instead of that array itself? To clarify, in this example

    >>> block([[One, Two],
    ...        [Three],
    ...        [four],

why not write that as


    >>> block([[One, Two],
    ...        Three,
    ...        four,

?